docs_new/features/core/output_types.md

```diff
@@ -0,0 +1,173 @@
+print(casted_result) # name=Aurora birth_date=datetime.date(1990, 6, 15) skills=['Stealth', 'Diplomacy']
+```
+
+## Output Type Categories
+
+We can group possible output types in several categories based on the use case they correspond to. While most of those types are native python or types coming from well-known third-party libraries, there are three Outlines-specific types: `JsonSchema`, `Regex` and `CFG`. Their use is explained below.
+
+### Basic Types
+
+The most straightfoward form of structed generation is to return an answer that conforms to a given basic type such as an int or a python list. You can use the basic Python types, but also the types contained in the `typing` library and combine them for more complex use cases. Some examples of valid Outlines output types:
+
+```python
+from typing import Dict, List, Optional, Tuple, Union
+
+output_type = float
+output_type = Dict[int, str]
+output_type = Tuple[Union[str, Dict[int, float]], List[bool], Optional[int]]
+```
+
+### Multiple Choice
+
+You can use your model to do classification by using the `Literal` or `Enum` output types. For instance:
+
+```python
+from typing import Literal
+from enum import Enum
+
+class PizzaOrBurger(Enum):
+    pizza = "pizza
+    burger = "burger
+
+# Equivalent multiple-choice output types
+output_type = Literal["pizza", "burger"]
+output_type = PizzaOrBurger
+```
+
+### Structured Objects
+
+What we call structured object here are roughly all common types that can be turned into a JSON schema. That includes:
+- A Pydantic class
+- A Dataclass
+- A TypedDict
+- A GenSon SchemaBuilder
+- A Callable (the parameters are turned into the keys and the type hinting is used to define the types of the values)
+
+For instance:
+
+```python
+from dataclasses import dataclass
+
+@dataclass
+class Character:
+    name: str
+    age: int
+
+ouput_type = Character
+
+def character(name: str, age: int):
+    return None
+
+output_type = character
+```
+
+On top of those five types that can directly be used with a model, there are also two other json schema formats for which we need to use an Outlines-specific class: JSON schema strings and dicts. As those do not have a distinct type, we must wrap them into an `outlines.types.JsonSchema` object. For instance:
+
+```python
+from outlines.types import JsonSchema
+
+schema_string = '{"type": "object", "properties": {"answer": {"type": "number"}}}'
+output_type = JsonSchema(schema_string)
+
+schema_dict = {
+    "type": "object",
+    "properties": {
+        "answer": {"type": "number"}
+    }
+}
+output_type = JsonSchema(schema_dict)
+```
+
+### Regex Patterns
+
+You can ask your model to generate text that follows a regex pattern. As the latter are expressed as simple raw string literals, we need to use an Outlines-specific class. Thus, the regex string should be be wrapped in an `outlines.types.Regex` object. For instance:
```

Comment:
```suggestion
Outlines provide support for text generation constrained by regular expressions. Since regular expressions are expressed as simple raw string literals, regex strings must wrapped in an `outlines.types.Regex` object to clarify the expected return type.
```